cmd/utils/nodecmd/chaincmd.go

```diff
@@ -160,117 +139,75 @@ func initGenesis(ctx *cli.Context) error {
 		dbc := &database.DBConfig{Dir: name, DBType: dbtype, ParallelDBWrite: parallelDBWrite,
 			SingleDB: singleDB, NumStateTrieShards: numStateTrieShards,
 			LevelDBCacheSize: 0, OpenFilesLimit: 0, DynamoDBConfig: dynamoDBConfig}
-		chaindb := stack.OpenDatabase(dbc)
+		chainDB := stack.OpenDatabase(dbc)
+
 		// Initialize DeriveSha implementation
 		blockchain.InitDeriveSha(genesis.Config.DeriveShaImpl)
 
-		_, hash, err := blockchain.SetupGenesisBlock(chaindb, genesis, params.UnusedNetworkId, false, overwriteGenesis)
+		_, hash, err := blockchain.SetupGenesisBlock(chainDB, genesis, params.UnusedNetworkId, false, overwriteGenesis)
 		if err != nil {
-			log.Fatalf("Failed to write genesis block: %v", err)
+			logger.Crit("Failed to write genesis block", "err", err)
 		}
-		logger.Info("Successfully wrote genesis state", "database", name, "hash", hash.String())
 
-		chaindb.Close()
+		// Write governance items to database
+		if err := governance.NewGovernance(genesis.Config, chainDB).WriteGovernance(0, govSet,
+			governance.NewGovernanceSet()); err != nil {
+			logger.Crit("Failed to write governance items", "err", err)
+		}
+
+		logger.Info("Successfully wrote genesis state", "database", name, "hash", hash.String())
+		chainDB.Close()
 	}
 	return nil
 }
 
-func getGovernanceItemsFromGenesis(genesis *blockchain.Genesis) governance.GovernanceSet {
-	g := governance.NewGovernanceSet()
+func ValidateGenesisConfig(g *blockchain.Genesis) error {
```

Comment:
`Enhance` and `ValidateGenesisConfig` do similar thing. How about combining them?

Reply:
I separated `Enhance` to make it more visual in the code since it forcibly updates users' configuration. It may look a little bit inefficient, however`config.Enhance` is called just once in common cases.
Thanks for the suggestion.

Reply:
How about combining the two functions and add comments instead?
I don't get what your point. Both functions are called once here and seperating the functions will not only make caution but also make the code dirty.